R/bin_spatial.R

```diff
@@ -0,0 +1,220 @@
+#' Assign fossil occurrences to spatial bins
+#'
+#' A function to assign fossil occurrences to spatial bins/samples using a
+#' hexagonal equal-area grid, or a distance-based approach.
+#'
+#' @param occdf \code{dataframe}. A dataframe of fossil occurrences you
+#' wish to bin. This dataframe should contain the decimal degree coordinates of
+#' your occurrences, and they should be of class `numeric`.
+#' @param lng \code{character}. The name of the column you wish to be treated
+#' as the input longitude (e.g., "lng" or "p_lng").
+#' @param lat \code{character}. The name of the column you wish to be treated
+#' as the input latitude (e.g., "lat" or "p_lat").
+#' @param method \code{character}. The desired method for spatially binning
+#' occurrence data. Either: "grid" or "dist". The "grid" method (default)
+#' spatially bins data into equal-area hexagonal grid cells based on a
+#' user-defined spacing between the center of adjacent cells. The "dist" method
+#' generates spatial samples for each locality (i.e., unique coordinate pairs)
+#' based on a user-defined distance. See details for further description.
+#' @param dist \code{numeric}. Under the "grid" method, this value indicates
+#' the desired spacing between the center of adjacent cells. Under the "dist"
+#' method, this is the distance threshold for defining spatial samples of
+#' occurrences. Note: `dist` should be provided in kilometres.
+#' @param buffer \code{numeric}. The buffer distance for defining unique
+#' localities (only useful if `method` = "dist").
+#' This argument allows the user to modify the definition of unique
+#' localities to draw from a larger area than point coordinates. This might be
+#' desirable if a high density of occurrences are in close proximity, but differ
+#' slightly in their coordinates (e.g., < 1 kilometre).
+#' Note: `buffer` should be provided in kilometres. The default value is `NULL`,
+#' which skips the buffer implementation.
+#' @param return \code{logical}. Should the equal-area grid information be
+#'  returned? Only useful if the method argument is set to "grid".
+#'  The default is `FALSE.`
+#'
+#' @return If the `method` argument is specified as "grid" and `return` as
+#' `FALSE`, a dataframe is returned of the original input `occdf` with
+#' cell information: ID number (`cell_ID`) and cell centroid coordinates
+#' (`cell_centroid_lng` and `cell_centroid_lat`). If `return` is
+#' set to `TRUE`, a list is returned with both the input `occdf` and grid
+#' information.
+#' If the `method` argument is specified as "dist", a list is returned of
+#' spatial samples drawn around unique localities based on the
+#' user-defined `dist`. Each element of the list contains occurrences within
+#' the geographic distance of the reference locality defined by the user.
```

Comment:
Was it a conscious decision to output the data in this way? Almost any downstream analysis will require these data frames to be unlisted into a single frame (as the 'grid' output is), maybe it would be better, as well as more consistent, to do this.

Reply:
Dropped, resolving.